File: backend/platform_app/models/lead.py
"""
Lead Model — Simplified

Core entity representing a discovered business lead.
Focused on contact discovery and export — no CRM, scoring, or AI analysis.
"""

from sqlalchemy import String, Text, ForeignKey, Integer, Float, JSON
from sqlalchemy.orm import relationship, Mapped, mapped_column
from platform_app.models.base import Base, TimestampMixin, generate_uuid


class Lead(Base, TimestampMixin):
    """A discovered business lead."""
    __tablename__ = "leads"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=generate_uuid)
    company_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("companies.id"), nullable=False, index=True
    )
    batch_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("lead_batches.id", ondelete="CASCADE"),
        nullable=False, index=True
    )

    # Business identity
    business_name: Mapped[str] = mapped_column(String(500), nullable=False)
    contact_name: Mapped[str | None] = mapped_column(String(500), nullable=True)
    designation: Mapped[str | None] = mapped_column(String(255), nullable=True)

    # Contact
    emails: Mapped[list[str]] = mapped_column(JSON, default=list)
    phones: Mapped[list[str]] = mapped_column(JSON, default=list)

    # Web presence
    website: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    linkedin_url: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    instagram_url: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    facebook_url: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    x_url: Mapped[str | None] = mapped_column(String(1000), nullable=True)

    # Location
    address: Mapped[str | None] = mapped_column(String(500), nullable=True)
    city: Mapped[str | None] = mapped_column(String(255), nullable=True)
    state: Mapped[str | None] = mapped_column(String(255), nullable=True)
    country: Mapped[str | None] = mapped_column(String(255), nullable=True)

    # Discovery metadata
    matched_keyword: Mapped[str | None] = mapped_column(String(500), nullable=True)
    source_primary: Mapped[str | None] = mapped_column(String(100), nullable=True)
    source_url: Mapped[str | None] = mapped_column(String(1000), nullable=True)

    # Ratings
    rating: Mapped[float | None] = mapped_column(Float, nullable=True)
    review_count: Mapped[int | None] = mapped_column(Integer, nullable=True)

    # Quality
    quality_status: Mapped[str] = mapped_column(
        String(50), default="valid", nullable=False,
        comment="valid, suspect, invalid"
    )
    verification_status: Mapped[str] = mapped_column(
        String(50), default="unverified", nullable=False,
        comment="unverified, verified, failed"
    )

    # Deduplication
    dedupe_key: Mapped[str | None] = mapped_column(String(500), nullable=True, index=True)

    # Relationships
    batch = relationship("LeadBatch", back_populates="leads")
    sources = relationship(
        "LeadSource", back_populates="lead",
        lazy="selectin", cascade="all, delete-orphan"
    )

File: frontend/src/types/lead.ts
export interface LeadBatch {
  id: string;
  name: string;
  location: string;
  requested_count: number;
  raw_results_count: number;
  duplicate_count: number;
  rejected_count: number;
  final_count: number;
  status: 'pending' | 'running' | 'partial' | 'completed' | 'failed' | 'cancelled';
  reason: string | null;
  keywords: string[];
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface Lead {
  id: string;
  batch_id: string;
  business_name: string;
  contact_name: string | null;
  designation: string | null;
  emails: string[];
  phones: string[];
  website: string | null;
  linkedin_url: string | null;
  instagram_url: string | null;
  facebook_url: string | null;
  x_url: string | null;
  address: string | null;
  city: string | null;
  state: string | null;
  country: string | null;
  matched_keyword: string | null;
  source_primary: string | null;
  source_url: string | null;
  rating: number | null;
  review_count: number | null;
  quality_status: 'valid' | 'suspect' | 'invalid';
  verification_status: 'unverified' | 'verified' | 'failed';
  created_at: string;
  updated_at: string;
}

export interface ExportJob {
  id: string;
  export_type: 'csv' | 'xlsx';
  selection_type: 'single_batch' | 'selected_batches' | 'all';
  lead_count: number;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  file_name: string | null;
  created_at: string;
}

export interface DashboardStats {
  total_leads: number;
  total_batches: number;
  last_search: string | null;
  unique_emails: number;
  unique_phones: number;
}

export interface SearchProgress {
  batch_id: string;
  status: 'running' | 'completed' | 'failed' | 'partial' | 'stopped';
  current_keyword: string | null;
  keywords_completed: number;
  keywords_total: number;
  leads_found: number;
  leads_target: number;
  duplicates?: number;
  scanned?: number;
  message: string;
  reason?: string;
  retry_round?: number;
}
